GitUI/Plugin/PluginRegistry.cs

```diff
@@ -73,7 +73,10 @@ public static void Register(IGitUICommands gitUiCommands)
 
             lock (Plugins)
             {
-                Plugins.ForEach(p => p.Register(gitUiCommands));
+                Plugins.ForEach(x => x.SettingsContainer.SetSettingsSource(gitUiCommands.GitModule.GetEffectiveSettings()));
+
+                Plugins.OfType<ILoadHandler>()
+                    .ForEach(x => x.OnLoad(gitUiCommands));
```

Comment:
We enumerating the collection multiple times, as well as performing casts. This is inefficient.

Reply:
Plugins list is of type List<IGitPlugin>. Collection in memory.
I don’t think that going through the collection several times in memory at this point is critical, we can optimize it if necessary, sacrificing the readability.
Are you suggesting replacing this with "foreach" and "is"?

Reply:
We should iterate in one go:
```
foreach (IPlugin plugin in Plugins)
{
    plugin.SettingsContainer.SetSettingsSource(gitUiCommands.GitModule.GetEffectiveSettings());

    plugin.OnLoad(gitUiCommands);

    // ...
}
```